### PR TITLE
[Alerting] Support set http_proxy config on Rancher notifier

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
Not support http_proxy setting now
Solution:
Add http_proxy configure for slack, pagerduty, wechat, webhook

Issue:
https://github.com/rancher/rancher/issues/17885

Types:
https://github.com/rancher/types/pull/855

Test:
https://github.com/rancher/rancher/pull/20743

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: pkg/controllers/user/alert/config/notifiers.go [hunk 1]
```diff
@@ -430,3 +446,107 @@ func (c *PushoverConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 	}
 	return checkOverflow(c.XXX, "pushover config")
 }
+
+// HTTPClientConfig configures an HTTP client.
+type HTTPClientConfig struct {
```
Comment (mrajashree) [inline pkg/controllers/user/alert/config/notifiers.go]: This and TLSConfig type and the method to generate http config from that seem to be exactly like the ones in [alertmanager](https://github.com/prometheus/alertmanager/pull/1222/files#diff-fbe65e96815b1559a386cd8521b5c79eR66) repo. Do you think we can vendor in and use them?

File: pkg/controllers/user/alert/configsyncer/configsyncer.go [hunk 2]
```diff
@@ -461,6 +462,17 @@ func (d *ConfigSyncer) addRecipients(notifiers []*v3.Notifier, receiver *alertco
 					ServiceKey:  alertconfig.Secret(notifier.Spec.PagerdutyConfig.ServiceKey),
 					Description: `{{ template "rancher.title" . }}`,
 				}
+
+				if notifier.Spec.PagerdutyConfig.HTTPClientConfig != nil {
+					url, err := toAlertManagerURL(notifier.Spec.PagerdutyConfig.HTTPClientConfig.ProxyURL)
+					if err != nil {
+						logrus.Errorf("Parse pageduty proxy url %s failed, %v", notifier.Spec.PagerdutyConfig.HTTPClientConfig.ProxyURL, err)
```
Comment (mrajashree) [inline pkg/controllers/user/alert/configsyncer/configsyncer.go]: nit: there's a typo in pagerduty spelling in this log (`pageduty`). Also can we change the error message to something like "Failed to parse pagerduty proxy url", for this and the other notifiers as well?

File: pkg/controllers/user/alert/config/notifiers.go (hunk 1 above)
Comment (aiwantaozi) [reply pkg/controllers/user/alert/config/notifiers.go]: Hi @mrajashree alertmanager also include other go package in that notifier and config types, like github.com/go-kit/kit, we need to vendor alertmanager, go-kit, and the package go-kit dependence, so we directly copy the go struct. If we really want to vendor alertmanager, we can update in a separate pr, this PR should be easier to test.

File: pkg/controllers/user/alert/configsyncer/configsyncer.go (hunk 2 above)
Comment (aiwantaozi) [reply pkg/controllers/user/alert/configsyncer/configsyncer.go]: Thanks, error message updated

**Pull request merged**